lib/rubocop/ext/regexp_node.rb

```diff
@@ -19,13 +19,19 @@ def assign_properties(*)
         super
 
         str = with_interpolations_blanked
-        @parsed_tree = begin
-          Regexp::Parser.parse(str, options: options)
+        begin
+          @parsed_tree = Regexp::Parser.parse(str, options: options)
         rescue StandardError
-          nil
+          @parsed_tree = nil
+        else
+          origin = loc.begin.end
+          source = @parsed_tree.to_s
+          @parsed_tree.each_expression(true) do |e|
+            e.origin = origin
+            # ts is a byte indice; convert it to a character indice
+            e.ts = source.byteslice(0, e.ts).size
```

Comment:
Could we do this lazily please instead?

Maybe simply adding a new method to the Nodes, and modifying the `expression` and `loc` methods?

Reply:
I added 2a043d4.  I'll squash if it looks okay.